pkg/standalone/runfileconfig/run_file_config_parser.go

```diff
@@ -0,0 +1,149 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runfileconfig
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"reflect"
+
+	"github.com/dapr/cli/utils"
+
+	"gopkg.in/yaml.v2"
+)
+
+// constants for the keys from the yaml file.
+const APPS = "apps"
+
+func (a *RunFileConfig) ParseAppsConfig(runFilePath string) error {
+	var err error
+	bytes, err := utils.ReadFile(runFilePath)
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal(bytes, &a)
+	if err != nil {
+		return fmt.Errorf("error in parsing the provided app config file: %w", err)
+	}
+	return nil
+}
+
+// ValidateRunConfig validates the run file config for mandatory fields and valid paths.
+func (a *RunFileConfig) ValidateRunConfig(runFilePath string) error {
+	baseDir, err := filepath.Abs(filepath.Dir(runFilePath))
+	if err != nil {
+		return fmt.Errorf("error in getting the absolute path of the provided run template file: %w", err)
+	}
+	if a.Version == 0 {
+		return errors.New("required field 'version' not found in the provided run template file")
+	}
+
+	// resolve relative path to absolute and validate all paths in commons.
+	err = a.resolvePathToAbsAndValidate(baseDir, &a.Common.ConfigFile, &a.Common.ResourcesPath)
+	if err != nil {
+		return err
+	}
+	for i := 0; i < len(a.Apps); i++ {
+		if a.Apps[i].AppDirPath == "" {
+			return errors.New("required filed 'app_dir_path' not found in the provided app config file")
+		}
+		// resolve relative path to absolute and validate all paths for app.
+		err := a.resolvePathToAbsAndValidate(baseDir, &a.Apps[i].ConfigFile, &a.Apps[i].ResourcesPath, &a.Apps[i].AppDirPath)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// GetApps returns a list of apps with the merged values forthe keys from common section of the YAML file.
+func (a *RunFileConfig) GetApps(filePath string) ([]Apps, error) {
+	bytes, err := utils.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+	appsKeys, err := a.getAppsKeysFromYaml(bytes)
+	if err != nil {
+		return nil, err
+	}
+	if len(a.Apps) != len(appsKeys) {
+		return nil, errors.New("error in parsing the provided app config file to extract provided configurations keys")
+	}
+	sharedConfigType := reflect.TypeOf(a.Common.SharedRunConfig)
+	fields := reflect.VisibleFields(sharedConfigType)
+	// Iterate for each field in common(shared configurations).
+	for _, field := range fields {
+		val := reflect.ValueOf(a.Common.SharedRunConfig).FieldByName(field.Name)
+		// Iterate for each app's configurations.
+		for i := range a.Apps {
+			appVal := reflect.ValueOf(a.Apps[i].RunConfig.SharedRunConfig).FieldByName(field.Name)
+			_, ok := appsKeys[i][field.Name]
+			// apppVal is the default value for the type and "ok" is for checking if this default value is not set explicitly in the app's configuration.
```

Comment:
this statement and check can be removed, since `field.Name` will have format `AppProtocol`, where as the key in the yaml file will be of format `app_protocol`. 
Adding support for zero values when a common value is given can be done later. For now it is ok to skip it.